openconfig_pyang/plugins/openconfig.py

```diff
@@ -373,6 +376,12 @@ def setup_ctx(self, ctx):
         "OC_KEY_ARGUMENT_UNQUOTED", ErrorLevel.MINOR,
         "All key arguments of a list should be quoted (%s is not)")
 
+    # bad type was used for a leaf or typedef
+    error.add_error_code(
+      "OC_BAD_TYPE", ErrorLevel.MAJOR,
```

Comment:
do we want to discourage use of the types or really forbid them?  (i.e., error vs. warning) ?

Reply:
I think really forbid them. `empty` cannot be sent in telemetry, and `bits` just increases complexity where `enum` will do the same job -- i.e.,:

```
leaf foo-bits {
  type bits {
    bit one {
      position 0;
    }
    bit two {
      position 1;
    }
  }
}
```

Would be rendered as "one two" on the wire according to the canonical representation in 6020. It seems better (to me) to have:

```
leaf-list foo-bits {
  type enumeration {
    enum one {
       value 0;
     }
     enum two {
       value 1;
     }
  }
}
```

which has the advantage that we can represent this as either a JSON list, or an array in telemetry - and string parsing is avoided.  The two things convey exactly the same information.

We've avoided the use of `bits` through all the LSDB models, which would be the most useful place for them, since we have named bits there.

Reply:
SGTM -- I've opened an issue openconfig/models#608 to add to the style guide.